cfme/services/catalogs/ansible_catalog_item.py

```diff
@@ -0,0 +1,282 @@
+    BootstrapSwitch,
+    Button,
+    Input,
+    Tab
+)
+
+from cfme.services.catalogs.catalog_item import AllCatalogItemView
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigate_to, navigator, CFMENavigateStep
+from utils.update import Updateable
+from utils.wait import wait_for
+from . import ServicesCatalogView
+
+
+class BootstrapSelect(VanillaBootstrapSelect):
+    """BootstrapSelect widget for Ansible Playbook Catalog Item form.
+
+    BootstrapSelect widgets don't have `data-id` attribute in this form, so we have to override ROOT
+    locator.
+
+    """
+    ROOT = ParametrizedLocator('.//select[normalize-space(@name)={@id|quote}]/..')
+
+
+class AnsibleCatalogItemForm(ServicesCatalogView):
+    title = Text(".//span[@id='explorer_title_text']")
+    name = Input("name")
+    description = Input("description")
+    display_in_catalog = BootstrapSwitch(name="display")
+    catalog = BootstrapSelect("catalog_id")
+
+    @View.nested
+    class provisioning(Tab):  # noqa
+        repository = BootstrapSelect("provisioning_repository_id")
+        playbook = BootstrapSelect("provisioning_playbook_id")
+        machine_credential = BootstrapSelect("provisioning_machine_credential_id")
+        cloud_type = BootstrapSelect("provisioning_cloud_type")
+        hosts = Input("provisioning_inventory")
+        escalate_privilege = BootstrapSwitch("provisioning_become_enabled")
+        verbosity = BootstrapSelect("provisioning_verbosity")
+        use_exisiting = Checkbox(locator=".//label[normalize-space(.)='Use Existing']/input")
+        create_new = Checkbox(locator=".//label[normalize-space(.)='Create New']/input")
+        provisioning_dialog_id = BootstrapSelect("provisioning_dialog_id")
+        provisioning_dialog_name = Input(name="vm.provisioning_dialog_name")
+
+    @View.nested
+    class retirement(Tab):  # noqa
+        # TODO Somehow need to handle a modal window
+        copy_from_provisioning = Button("Copy from provisioning")
+        repository = BootstrapSelect("retirement_repository_id")
+        playbook = BootstrapSelect("provisioning_playbook_id")
+        machine_credential = BootstrapSelect("provisioning_machine_credential_id")
+        cloud_type = BootstrapSelect("provisioning_cloud_type")
+        hosts = Input("retirement_inventory")
+        escalate_privilege = BootstrapSwitch(name="provisioning_become_enabled")
+        verbosity = BootstrapSelect("retirement_verbosity")
+        remove_resources = BootstrapSelect("vm.catalogItemModel.retirement_remove_resources")
+
+    cancel = Button("Cancel")
+
+
+class SelectCatalogItemTypeView(ServicesCatalogView):
+    title = Text(".//span[@id='explorer_title_text']")
+    catalog_item_type = BootstrapSelect("st_prov_type", can_hide_on_select=True)
+    add = Button("Add")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer() and
```

Comment:
You sure about this one? This usually is a property. And are you sure in_explorer is an appropriate name for this property? Catalog items do not have an "Explorer" :)